src/analysis/openstreetmap/qgsosmdownload.cpp

```diff
@@ -20,11 +20,13 @@
 
 #include "qgsnetworkaccessmanager.h"
 #include "qgsrectangle.h"
+#include "qgssettings.h"
 
 
 QString QgsOSMDownload::defaultServiceUrl()
 {
-  return QStringLiteral( "http://overpass-api.de/api/interpreter" );
+  QgsSettings settings;
+  return settings.value( "overpass_url", "http://overpass-api.de/api/interpreter" ).toString();
```

Comment:
@Gustry this is correct usage of the API.
In this particular case, the API endpoint is not going to likely change over time, so I think that an inline default is appropriate.

I only suggest that instead of using a key in the global namespace "overpass_url" you could use one of the available sections: http://qgis.org/api/classQgsSettings.html#aba2f174ee000905cac8aff26665ed7c7
The call could then be:
```
return settings.value( "overpass_url", "http://overpass-api.de/api/interpreter", QgsSettings::Section::Plugins);
```
In this case, the setting key would automatically be created as "plugins/overpass_url", to be even more specific, you could use "analysis/overpass_url" as a key, that would translate in "plugins/analysis/overpass_url".
See: http://qgis.org/api/classQgsSettings.html#a15948771a3ef0280d6df1938eefab2fe